keps/sig-node/2400-node-swap/README.md

```diff
@@ -441,6 +469,25 @@ Additionally, end user may decide to disable swap completely for a Pod or a cont
 In the early release of this feature, there was a goal to support cgroup v1. As the feature progressed, sig-node realized that supporting swap with cgroup v1 would be very difficult.
 Therefore, this feature is limited to cgroupv2 only. The main goal is to deprecate cgroupv1 eventually so this should not be a major inconvience.
 
+#### Memory-backed volumes
+
+Kubernetes guarantees that some volumes' memory would never reside on disk, e.g. Secrets, memory-backed emptyDirs, etc.
+Behind the scenes, Kubelet mounts such volumes as tmpfs volumes on the host.
+
+To address this risk, if `--fail-swap-on=false`, the [tmpfs noswap option](https://www.kernel.org/doc/html/latest/filesystems/tmpfs.html)
+will be used in order to prevent the volumes' pages from swapping to disk.
+
+Bear in mind that the tmpfs noswap option is fairly new and is supported in kernel versions >= 6.4. However, different
+Linux distributions can decide to backport this options to older versions of the kernel. Therefore, when
+`--fail-swap-on=false` is being provided on a node:
+* If the kernel version equals or is above 6.4, the tmpfs noswap option is being used when necessary.
+* Else, kubelet would try to mount a dummy volume with the tmpfs noswap option to understand whether the option is
+  backported. If the mount succeeds, the tmpfs noswap option is being used when necessary.
+* Else, kubelet would raise a warning log entry about the option not being supported and the possible risk.
```

Comment:
but it will mount it anyway, right?

Reply:
AFAIK kubelet will fail to mount if the kernel does not support this feature. So it won’t be mounted in the “correct” way due to a limitation of the kernel.

Reply:
> but it will mount it anyway, right?

Not sure I understand, mount what?
This "dummy mount" will fail if the tmpfs noswap is not supported. It is basically a method to determine whether the noswap option is supported or not. If it is supported, it'll be used to mount memory-backed volumes. If not, a warning log entry will be raised, and memory-backed volumes will be mounted without the noswap option.

Hope this answers your question, let me know if you need more info